src/internal/clusterstate/v2.6.0/pfsdb.go

```diff
@@ -111,59 +61,107 @@ func validateExistingDAGs(cis []*v2_5_0.CommitInfo) error {
 }
 
 func removeAliasCommits(ctx context.Context, tx *pachsql.Tx) error {
-	var oldCIs []*v2_5_0.CommitInfo
-	var err error
-	// TODO(provenance): this function is doing more than just removing aliases
-	// first fill commit provenance to all the existing commit sets
-	if oldCIs, err = listCollectionProtos(ctx, tx, "commits", &v2_5_0.CommitInfo{}); err != nil {
-		return errors.Wrap(err, "populate the pfs.commits table")
-	}
-	for i, ci := range oldCIs {
-		log.Info(ctx, "add old commit to pfs.commits",
-			zap.String("commit", fmt.Sprintf("%s@%s=%s",
-				repoKey(ci.Commit.Branch.Repo),
-				ci.Commit.Branch.GetName(),
-				ci.Commit.Id)),
-			zap.String("progress", fmt.Sprintf("%v/%v", i, len(oldCIs))),
-		)
-		if err := addCommit(ctx, tx, ci.Commit); err != nil {
-			return err
-		}
+	cis, err := listCollectionProtos(ctx, tx, "commits", &v2_5_0.CommitInfo{})
+	if err != nil {
+		return err
 	}
-	oldCIMap := make(map[string]struct{})
-	for _, ci := range oldCIs {
-		oldCIMap[oldCommitKey(ci.Commit)] = struct{}{}
+	type commit struct {
+		id   int
+		info *v2_5_0.CommitInfo
 	}
-	deleteCommits := make(map[string]*v2_5_0.CommitInfo)
-	for i, ci := range oldCIs {
-		log.Info(ctx, "branch provenance for commit",
-			zap.String("commit", oldCommitKey(ci.Commit)),
-			zap.Any("branch provenance", ci.DirectProvenance),
-			zap.String("progress", fmt.Sprintf("%v/%v", i, len(oldCIs))),
-		)
-		for j, b := range ci.DirectProvenance {
-			log.Info(ctx, "add commit provenance",
-				zap.String("from", oldCommitKey(ci.Commit)),
-				zap.String("to", oldCommitKey(b.NewCommit(ci.Commit.Id))),
-				zap.String("progress", fmt.Sprintf("%v/%v", j, len(ci.DirectProvenance))),
-			)
-			// if the provenant commit's repo was deleted, it's reference may
-			// still exist in the old provenance, so we skip mapping it over to the new model
-			if _, ok := oldCIMap[oldCommitKey(b.NewCommit(ci.Commit.Id))]; !ok {
+	commits := make(map[string]*commit)
+	id := 1
+	for _, ci := range cis {
+		c := &commit{info: ci}
+		if ci.Origin.Kind != 4 {
+			c.id = id
```

Comment:
what assurance do we have that makes us confident we can rely on ids reflecting insertion order?

Reply:
I refactored this to pull from the `pfs.commits` table after insertion just to be safe here.